sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -498,12 +500,25 @@ class Dataset[T] private[sql](
    * @since 1.6.0
    */
   def explain(extended: Boolean): Unit = {
-    val explain = ExplainCommand(queryExecution.logical, extended = extended)
-    sparkSession.sessionState.executePlan(explain).executedPlan.executeCollect().foreach {
-      // scalastyle:off println
-      r => println(r.getString(0))
-      // scalastyle:on println
+    val qe = if (isStreaming) {
```

Comment:
let's add some comments to explain why we inline the implementation of explain command here.

Reply:
Can we put this into a method that is shared by both the `Dataset` and the explain command, e.g.:
```scala
def explainQueryExecution(plan: LogicalPlan, queryExecution: => QueryExecution): QueryExecution = {
  if (plan.isStreaming) {
      // This is used only by explaining `Dataset/DataFrame` created by `spark.readStream`, so the
      // output mode does not matter since there is no `Sink`.
      new IncrementalExecution(
        sparkSession, logicalPlan, OutputMode.Append(), "<unknown>",
        UUID.randomUUID, UUID.randomUUID, 0, OffsetSeqMetadata(0, 0))
    } else {
      queryExecution
    }
}
```

Reply:
Added comments and put common code into a method. Thanks.